### PR TITLE
Enable coverage reports

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

File: bin/install_travis.sh [hunk 1]
```diff
@@ -14,25 +14,35 @@ if [[ "${CC}" == "" ]]; then
...
         export CC=gcc-4.7
         export CXX=g++-4.7
+        export GCOV_EXECUTABLE=gcov-4.7
     fi
 fi
 
+if [[ "${WITH_COVERAGE}" == "yes" ]]; then
+    pip install --user cpp-coveralls
+fi
+
```
Comment (certik) [inline bin/install_travis.sh]: This seems to be executed again in `test_travis.sh`, so it is redundant here?

Comment (isuruf) [inline bin/install_travis.sh]: Right. I'll remove it from `test_travis.sh`

**Comment by certik** [comment]
Otherwise +1.

File: CMakeLists.txt [hunk 2]
```diff
@@ -372,6 +372,10 @@ if (CMAKE_BUILD_TYPE STREQUAL "Debug")
     set(WITH_SYMENGINE_ASSERT yes) # Also enable assertions
 endif()
 
+if (WITH_COVERAGE)
+    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fprofile-arcs -ftest-coverage")
```
Comment (certik) [inline CMakeLists.txt]: I think here you can just use `-coverage`, as suggested by @bluescarni.

**Comment by certik** [comment]
(I accidentally hit close.)

**Comment by isuruf** [comment]
This PR is ready now

**Comment by certik** [comment]
That looks good, thanks for enabling this!

**Pull request merged**